pkg/cli/options.go

```diff
@@ -171,26 +171,44 @@ func parseExternalPluginArgs() (args []string) {
 	return args
 }
 
+// isHostSupported checks whether the host system is supported or not.
+func isHostSupported(host string) bool {
+	for _, platform := range supportedPlatforms {
+		if host == platform {
+			return true
+		}
+	}
+	return false
+}
+
 // getPluginsRoot detects the host system and gets the plugins root based on the host.
 func getPluginsRoot(host string) (pluginsRoot string, err error) {
+	if !isHostSupported(host) {
+		// freebsd, openbsd, windows...
+		return "", fmt.Errorf("host not supported: %v", host)
+	}
+
+	pluginsRelativePath := filepath.Join("kubebuilder", "plugins")
+	if xdgHome := os.Getenv("XDG_CONFIG_HOME"); xdgHome != "" {
+		return filepath.Join(xdgHome, pluginsRelativePath), nil
+	}
+
 	switch host {
 	case "darwin":
 		logrus.Debugf("Detected host is macOS.")
-		pluginsRoot = filepath.Join("Library", "Application Support", "kubebuilder", "plugins")
+		pluginsRoot = filepath.Join("Library", "Application Support", pluginsRelativePath)
 	case "linux":
 		logrus.Debugf("Detected host is Linux.")
-		pluginsRoot = filepath.Join(".config", "kubebuilder", "plugins")
-	default:
-		// freebsd, openbsd, windows...
-		return "", fmt.Errorf("Host not supported: %v", host)
+		pluginsRoot = filepath.Join(".config", pluginsRelativePath)
 	}
-	userHomeDir, err := getHomeDir()
+
+	userHomeDir, err := os.UserHomeDir()
 	if err != nil {
 		return "", fmt.Errorf("error retrieving home dir: %v", err)
 	}
 	pluginsRoot = filepath.Join(userHomeDir, pluginsRoot)
 
-	return pluginsRoot, nil
```

Comment:
This method expects a return `(pluginsRoot string, err error) ` I think it is missing return it

Reply:
the reason I have replaced `return pluginsRoot, nil` with just `return` is because the function uses named returns so I think specifying the variables in the return is redundant, but I can add it back (maybe because of a style convention?) just let me know.
thank you!